### PR TITLE
CLN: clean Apply._try_aggregate_string_function

### PR DESCRIPTION
**Pull request by topper-123**
CC @rhshadrach.

### CODE REVIEW

File: pandas/core/apply.py [hunk 1]
```diff
@@ -601,34 +601,32 @@ def normalize_dictlike_arg(
...
+        if hasattr(obj, arg):
+            f = getattr(obj, arg)
             if callable(f):
                 return f(*args, **kwargs)
 
-            # people may try to aggregate on a non-callable attribute
+            # people may aggregate on a non-callable attribute
             # but don't let them think they can pass args to it
             assert len(args) == 0
             assert len([kwarg for kwarg in kwargs if kwarg not in ["axis"]]) == 0
```
Comment (rhshadrach) [inline pandas/core/apply.py:621]: I haven't looked into this, but at least at the surface it seems like users may be able to generate an AssertionError here? Perhaps these should be changed to ValueErrors (also okay to not tackle here).

File: pandas/core/apply.py [hunk 2]
```diff
@@ -601,34 +601,32 @@ def normalize_dictlike_arg(
...
+        - try to find a function (or attribute) on obj
         - try to find a numpy function
         - raise
         """
         assert isinstance(arg, str)
 
-        f = getattr(obj, arg, None)
-        if f is not None:
+        if hasattr(obj, arg):
+            f = getattr(obj, arg)
```
Comment (rhshadrach) [inline pandas/core/apply.py:614]: Not sure I see the value of this change (but really, not much opposition to it either). My understanding of `hasattr` is that it is essentially:

```
try:
    getattr(obj, arg)
    return True
except AttributeError:
    return False
```

Now we're calling getattr twice whereas before it was only once.

File: pandas/core/apply.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/core/apply.py:621]: Confirmed:

```
ser = pd.Series([1, 1, 2])
print(ser.agg('size', a=1))
# File "pandas/pandas/core/apply.py", line 621, in _try_aggregate_string_function
#     assert len([kwarg for kwarg in kwargs if kwarg not in ["axis"]]) == 0
# AssertionError
```

File: pandas/core/apply.py (hunk 2 above)
Comment (topper-123) [reply pandas/core/apply.py:614]: Ok, I'm mostly thinking about code organization (lining up the if/elif/else), though if there is a performance difference, it looks like it's max around some 10s of nanoseconds, i.e. negligible.

File: pandas/core/apply.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/apply.py:621]: Yeah, I though about fixing that, but then it wouldn't be just a clean-up. I can do it in a follow-up?

File: pandas/core/apply.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/core/apply.py:614]: > Ok, I'm mostly thinking about code organization (lining up the if/elif/else)

Early return is also a viable strategy:

```
def foo(x):
    if x == 2:
        return 3
    ... rest of method ...
```

as opposed to 

```
def foo(x):
    if x == 2:
        return 3
    else:
        ... rest of method ...
```

Early return keeps the indentation level lower and can make complex methods more readable.

That isn't to say I'm opposed to the changes here, but rather it seems to me someone can come along next week and undo this and I wouldn't be opposed to that either. Similar with the getattr/hasattr.

Comment (topper-123) [reply pandas/core/apply.py:614]: Ok, preferences can differ, no big deal. If I revert the code inside the method, would the PR then be ok?

Comment (rhshadrach) [reply pandas/core/apply.py:614]: Yes - and if you felt strongly about these changes I wouldn't object.

Comment (topper-123) [reply pandas/core/apply.py:614]: Ok, if its not a big issue wuth you eithrer, I would prefer then to get it in as-is, just to save the time rewriting...

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
@topper-123 - been a while since the tests ran, so just being cautious (paranoid? :laughing:) and wanted to kick them off again. Will merge on green.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @topper-123

**Pull request merged**